db/revtree.go

```diff
@@ -395,3 +405,25 @@ func encodeRevisions(revs []string) Body {
 	}
 	return Body{"start": start, "ids": ids}
 }
+
+// Given a revision history encoded by encodeRevisions() and a list of possible ancestor revIDs,
+// trim the history to stop at the first ancestor revID. If no ancestors are found, trim to
+// length maxUnmatchedLen.
+func trimEncodedRevisionsToAncestor(revs Body, ancestors []string, maxUnmatchedLen int) bool {
+	start, digests := splitRevisionList(revs)
+	if digests == nil {
+		return false
```

Comment:
It returns false if it fails, i.e. there's an error unpacking the revision list from `revs`.

Since this whole thing is just to optimize the size of the response, it didn't seem worthwhile to make a failure here cause the whole request to fail, which is why the caller ignores the return value.